// mqtt/build/lib/PingTimer.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const worker_timers_1 = require("worker-timers");
const is_browser_1 = __importDefault(require("./is-browser"));
class PingTimer {
    constructor(keepalive, checkPing) {
        this.setTimeout = is_browser_1.default ? worker_timers_1.setTimeout : setTimeout;
        this.clearTimeout = is_browser_1.default ? worker_timers_1.clearTimeout : clearTimeout;
        this.keepalive = keepalive * 1000;
        this.checkPing = checkPing;
        this.setup();
    }
    setup() {
        this.timer = this.setTimeout(() => {
            this.checkPing();
            this.reschedule();
        }, this.keepalive);
    }
    clear() {
        if (this.timer) {
            this.clearTimeout(this.timer);
            this.timer = null;
        }
    }
    reschedule() {
        this.clear();
        this.setup();
    }
}
exports.default = PingTimer;
